dss-api.yml

```diff
@@ -119,6 +119,59 @@ paths:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
+    put:
+      summary: Create a file
+      description: |
+        Create a new file with a given UUID.  The file content is passed in through a cloud URL.  The file on the cloud
+        provider must have the file checksums set as metadata.
+      parameters:
+        - name: uuid
+          in: path
+          description: A RFC4122-compliant ID for the file.
+          required: true
+          type: string
+        - name: extras
+          in: body
+          required: true
+          schema:
+            type: object
+            properties:
+              source_url:
+                description: Cloud URL for source data.
+                type: string
+                pattern: "^(gs|s3|wasb)://"
+              bundle_uuid:
+                description: A RFC4122-compliant ID for the bundle that contains this file.
+                type: string
+              creator_uid:
+                description: User ID who is creating this file.
+                type: integer
+                format: int64
+              content_type:
```

Comment:
Consider leaving this as optional. The default would then be to copy the content-type of the object referred in source_url.